contributors: add benchmark guide (#157)
pingcap/community: contributors/README.md

TBD

### Benchmark

If the change affects TIDB's performance, the benchmark data is normally required in the description. You can use the [benchstat](https://godoc.org/golang.org/x/perf/cmd/benchstat) tool to format benchmark data for change descriptions.

> We could do is run the benchmarks over multiple runs ie

Reviewer: ```suggestion
The following script runs benchmark multiple times (10)
```

```bash
go test -run=^$ -bench=BenchmarkXXX -count=10
```

> **Tip**: To make the result more readable, you can copy and paste the output to two both the old.txt and new.txt, and then run: 

Reviewer: ```suggestion
**Tip**: To make the result more readable, you can copy and paste the output to both the old.txt and new.txt, and then run: 
```

> And we could make more clear and reportable differences by copying and pasting that output to two different files: old.txt and new.txt and then running: 

Reviewer: ```suggestion
**Tip**: To make the result more readable, you can copy and paste the output to two both the old.txt and new.txt, and then run:

```
benchstat old.txt new.txt
```

> and paste the result into your pull request description. Here is a good [example](https://github.com/pingcap/tidb/pull/12903#issue-331440170).

Reviewer: ```suggestion

Paste the result into your pull request description. Here is a good [example](https://github.com/pingcap/tidb/pull/12903#issue-331440170).
```

TBD

### Benchmark

> #### TiDB

Reviewer: ```suggestion
```

> It's appreciate to write benchmark code and add benchmark data if the change affects performance. The [benchstat](https://godoc.org/golang.org/x/perf/cmd/benchstat) tool is conventionally used to format benchmark data for change descriptions.

Reviewer: ```suggestion
If the change affects TIDB's performance, the benchmark data is normally required in the description. You can use the [benchstat](https://godoc.org/golang.org/x/perf/cmd/benchstat) tool to format benchmark data for change descriptions.
```

We could do is run the benchmarks over multiple runs ie

```bash

> go test -run=^$ -bench=BenchmarkXXX -count=10

Reviewer 2: which directory should we enter in order to run this command successfully?

Author: I think it's a common knowledge for go user.
